duckduckgo/duckduckgo-privacy-extension
# Use findParent() to identify a site as major tracker networks, and improve matching with tosdr ratings

**Reviewer:** @jdorweiler, @andrey-p 


## Description:
The extension uses loose regex matching for both tosdr privacy ratings and to identify when the mainframe site is a major tracker network.
This leads to grading bugs.

In this PR:
- To fix Tosdr, make the regex stricter by only matching from the beginning of the URL, and use tldjs.getDomain()
- Use the entityMap, and more specifically the findParent() function to identify whether the parent is a major tracker network
- Move the above mentioned findParent() to background/utils.es6.js (more suitable place now that it's not specific to trackers)

## Steps to test this PR:
1. Build and reload extension
2. Go to deletefacebook.com
3. The grade should be B and the scorecard should *not* have a line for "site is a major tracker network"; tosdr should be unknown
4. Go to youtube.com or any major tracker network-owned website with a different domain than the parent company
5. Grade is "D", and the scorecard lists it as a major tracker network; tosdr should be bad
6. All other sites should be unaffected
7. Run unit tests with npm run test, or npm run test-debug

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
